pkg/plugins/policies/meshretry/plugin/v1alpha1/plugin_test.go

```diff
@@ -587,6 +598,102 @@ var _ = Describe("MeshRetry", func() {
 			},
 			goldenFilePrefix: "http-real-mesh-service",
 		}),
+		Entry("http retry mesh external service", testCase{
```

Comment:
should we have one for tcp and grpc as well?

Reply:
I’m not entirely convinced, but we can add it. These tests will require some rewriting (similar to the ones you started in MeshHTTPRoute), as we mock a lot of data. Even if the test passes, it might not work in E2E. I think these tests are useful for checking the output Envoy config, but they don't test the entire flow.

Reply:
Is there any reason for this not to work for grpc, they are configured in a similar place right?

Reply:
I don’t think it shouldn’t. We can add another case with different protocol